rep_gen_test.py

```diff
@@ -0,0 +1,169 @@
+        os.mkdir(cls.tmp_dir)
+
+        # Build temporary config file
+        print("Building temporary source file {}.".format(cls.config_file))
+        with open(cls.config_file, 'w+') as f:
+            f.write('[GOOMBAS]\n')
+            f.write('TOAD = PRINCESS PEACH\n')
+
+    @classmethod
+    def tearDownClass(cls):
+        print("Deleting temporary directory and file for testing.")
+        shutil.rmtree(cls.tmp_dir)
+
+    def test_default_init(self):
+        """Test default initialization. When default initialization value
+        changes, or any private attribute does not match public attribute, this
+        test will fail."""
+        # Assert rg._config_path and rg._out_dir are initialized correctly
+        rg = RepresentationGenerator()
+        self.assertEqual(rg._config_path, 'configs/sample_config.ini')
+        self.assertEqual(rg._out_dir, 'embeddings')
+
+        # Assert DatasetBuilder and ModelBuilder is created during
+        # initialization.
+        mock_custom_dataset.DatasetBuilder.assert_called_with(
+            config_path='configs/sample_config.ini', one_hot=False)
+        mock_custom_model.ModelBuilder.assert_called_with(
+            config_path='configs/sample_config.ini'
+        )
+        # Assert that model builder is used for getting self._model
+        rg._model_builder.get_encoder.assert_called()
+        self.assertEqual(rg._model, rg._model_builder.get_encoder())
+
+    def test_config_path_setter(self):
+        # Test setter in initialization
+        rg = RepresentationGenerator(config_path=self.config_file)
+        self.assertEqual(rg._config_path, self.config_file)
+
+        # Test setter after initialization
+        rg = RepresentationGenerator()
+        rg.config_path = self.config_file
+        self.assertEqual(rg._config_path, self.config_file)
+
+        # Test exception
+        with self.assertRaises(ValueError):
+            RepresentationGenerator(os.path.join(self.config_file,'123'))
+        with self.assertRaises(ValueError):
+            rg.config_path = os.path.join(self.config_file,'123')
+
+    def test_out_dir_setter(self):
+        # Test setter in initialization
+        rg = RepresentationGenerator(out_dir=self.tmp_dir)
+        self.assertEqual(rg._out_dir, self.tmp_dir)
+
+        # Test setter after initialization
+        rg = RepresentationGenerator()
+        rg.out_dir = self.tmp_dir
+        self.assertEqual(rg._out_dir, self.tmp_dir)
+
+    def test_get_embeddings(self):
+        # Assert that dataset_builder and
+        rg = RepresentationGenerator()
+        # Get mocked filename dataset
+        mock_img = MagicMock()
+        mock_label = MagicMock()
+        mock_ds = [(mock_img, mock_label)]
+
+        # Mock function get_filename_dataset.
+        with patch.object(rg._dataset_builder, 'get_filename_dataset',
+                          return_value=mock_ds) as get_fd:
+            codepoints, embeddings = rg.get_embeddings(img_dir=self.tmp_dir)
```

Comment:
you should assert the codepoints, embeddings values returned by  get_embeddings are correct based on what the mock object would return. Take a look of side_effect  and return_value in https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect